components/tidb_query/src/util.rs

```diff
@@ -103,9 +105,24 @@ pub fn get_pk(col: &ColumnInfo, h: i64) -> Datum {
     }
 }
 
+/// `check_key_type` checks if the key is the type we want, `sep_type` should be
+/// `table::RECORD_PREFIX_SEP` or  `table::INDEX_PREFIX_SEP` .
+#[inline]
+pub fn check_key_type(key: &[u8], sep_type: &[u8]) -> Result<(), EvaluateError> {
+    if sep_type != &key[table::TABLE_PREFIX_KEY_LEN..table::PREFIX_LEN] {
+        Err(EvaluateError::Other("got a wrong key type".to_string()))
```

Comment:
It's better to attach extra information to the error message help us disgnose problem, e.g: `format!("Expected key type {:?}, but got {:?})", sep_type, &key[table::TABLE_PREFIX_KEY_LEN..table::PREFIX_LEN])`